Mtneer/mushu
# Csv upload

# Description
This PR includes a new feature which allows users to upload transaction history from .CSV files downloaded from their financial accounts. It provides users a preview of the csv data, requires them to categorize each transaction (if that info is not readily available), and saves the new transactions to the database. It also assigns these transactions to an account of their choosing.

## Type of change
Please delete options that are not relevant.

# Testing Instructions
- pull down new branch, start the react app, start the web api, and login
- from the transactions tab, select the "+ Add New" button
- select or drag&drop a .CSV file of financial transactions into the importer
- confirm the preview
- drag&drop the appropriate columns to the desired containers
- click confirm
- click finish
- select a category for each transaction with a missing category
- click the submit button at the top
- verify that you are navigated back to the list of transactions and that the new transactions appear

# Checklist:
